### ISSUE TITLE
Control code prolog, code epilog and code delimiter/separator 

### ISSUE THREAD

**Issue by PhilippSalvisberg** [oddgen/oddgen#34]
When selecting multiple objects to generate, oddgen in version 0.2 produces an empty line between the generated artefacts. This is for okay for PL/SQL, but it might be wrong when generating other target code (such as XML).

The current default behaviour is good, but the generator should have control over 
- the code separator/delimiter 
- the code prolog (before the first generated code) 
- the code epilog (after the last generated code). 

Prolog and epilog are now empty. Which is also a good default, but the generator should control that too.

PL/SQL and Java interface are to be extended accordingly.

**Comment by mortenbra** [comment]
I was just thinking about something similar today; that it should be possible to have some "pre text" and "post text" that gets generated only once before and after all objects selected for generation (ie only once, regardless of the number of selected objects).

The use case would be to generate some header and footer text for the whole script, such as PL/SQL compiler settings, standard comment blocks, package begin/end blocks for a set of procedures, etc.

Maybe this should be a separate issue, or perhaps change the name of this issue to include this?

**Comment by PhilippSalvisberg** [comment]
> Maybe this should be a separate issue, or perhaps change the name of this issue to include this?

From my point of view it is the same issue, so no need to add an additional one. Why do you think they are different?

**Comment by MikeKutz** [comment]
It may be useful for the prolog and epilog functions to have an array of what is being generated.  This way, the "header" of the script can describe what objects the script includes.

For the separator function, "before" and "after" objects might be needed.  Also, I can see the "generator parameters" being an "IN/OUT" parameter.  This would allow a parameter to act like a "counter".

Just a suggestion.

MK

**Comment by mortenbra** [comment]
> From my point of view it is the same issue, so no need to add an additional one. Why do you think they are different?

Yes, we are indeed talking about the same thing. Sorry for the confusion! :-)

**Comment by PhilippSalvisberg** [comment]
> Yes, we are indeed talking about the same thing. Sorry for the confusion! :-)

No problem.

**Comment by chicagozer** [comment]
prolog/epilog for the Java interface would be fantastic. In our use case we would like to code generate a single PKB with individual procedures referencing each object selected in oddgen.

**Comment by PhilippSalvisberg** [comment]
It is part of the [new Java interface](https://github.com/oddgen/oddgen/blob/issue_28/sqldev/src/main/java/org/oddgen/sqldev/generators/OddgenGenerator2.xtend) in the current development branch.

**Comment by PhilippSalvisberg** [comment]
is part of the new v0.3.0 interface, implemented with #28
